### PR TITLE
[cli] support default config without specifying a config file

### PR DESCRIPTION
**Pull request by wgtdkp**
This PR includes two small enhancements:
1. Support starting the `commissioner-cli` without specifying a configuration file.
    a. pskc is set to all-one.
    b. log messages are printed to syslog (/var/log/syslog).
2. Add sys logger support.

### CODE REVIEW

File: src/app/cli/main.cpp [hunk 1]
```diff
@@ -89,18 +89,25 @@ static void HandleSignalInterrupt()
 
 int main(int argc, const char *argv[])
 {
-    Error  error;
-    Config config;
+    Error       error;
```
Comment (bukepo) [inline src/app/cli/main.cpp:92]: may enhance to use getopt in future PRs.

**Review by bukepo — APPROVED** [review]
Thanks! :+1:
Is it possible to add a test or update existing tests without specifying the config file?

File: src/app/cli/main.cpp (hunk 1 above)
Comment (wgtdkp) [reply src/app/cli/main.cpp:92]: I see DSR's PR https://github.com/openthread/ot-commissioner/pull/202 already changed to use `getopt` so let that PR do the change.

File: tests/integration/test_joining.sh [hunk 2]
```diff
@@ -47,6 +47,26 @@ test_joining() {
...
+    ## enable all MeshCoP joiners
+    send_command_to_commissioner "joiner enable meshcop ${JOINER_EUI64} ${JOINER_CREDENTIAL}"
+
+    start_joiner "meshcop"
+
+    stop_commissioner
+
+    stop_daemon
+}
+
```
Comment (wgtdkp) [inline tests/integration/test_joining.sh:69]: @bukepo The test case for starting without a config file.

**Review by jwhui — APPROVED** [review]

**Pull request merged**